core/jobs/batch_jobs/suggestion_stats_computation_jobs.py

```diff
@@ -0,0 +1,296 @@
+
+datastore_services = models.Registry.import_datastore_services()
+
+
+class GenerateTranslationContributionStatsJob(base_jobs.JobBase):
+    """Job that indexes the explorations in Elastic Search."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Generates the translation contributins stats.
+
+        Returns:
+            PCollection. A PCollection of 'SUCCESS x' results, where x is
+            the number of generated stats..
+        """
+        suggestions_grouped_by_target = (
+            self.pipeline
+            | 'Get all non-deleted suggestion models' >> ndb_io.GetModels(
+                suggestion_models.GeneralSuggestionModel.get_all(
+                    include_deleted=False))
+            # We need to window the models so that CoGroupByKey below
+            # works properly.
+            | 'Window the suggestions' >> beam.WindowInto(
+                beam.window.Sessions(10 * 60))
+            | 'Filter translate suggestions' >> beam.Filter(
+                lambda m: (
+                    m.suggestion_type ==
+                    feconf.SUGGESTION_TYPE_TRANSLATE_CONTENT
+                ))
+            | 'Transform to suggestion domain object' >> beam.Map(
+                suggestion_services.get_suggestion_from_model)
+            | 'Group by target' >> beam.GroupBy(lambda m: m.target_id)
+        )
+        exp_opportunities = (
+            self.pipeline
+            | 'Get all non-deleted opportunity models' >> ndb_io.GetModels(
+                opportunity_models.ExplorationOpportunitySummaryModel.get_all(
+                    include_deleted=False))
+            # We need to window the models so that CoGroupByKey below
+            # works properly.
+            | 'Window the opportunities' >> beam.WindowInto(
+                beam.window.Sessions(10 * 60))
+            | 'Transform to opportunity domain object' >> beam.Map(
+                opportunity_services.
+                get_exploration_opportunity_summary_from_model)
+            | 'Group by ID' >> beam.GroupBy(lambda m: m.id)
+        )
+
+        new_user_stats_models = (
+            {
+                'suggestion': suggestions_grouped_by_target,
+                'opportunity': exp_opportunities
+            }
+            | 'Merge models' >> beam.CoGroupByKey()
+            | 'Get rid of key' >> beam.Values()  # pylint: disable=no-value-for-parameter
+            | 'Generate stats' >> beam.ParDo(
+                lambda x: self._generate_stats(
+                    x['suggestion'][0] if len(x['suggestion']) else [],
+                    x['opportunity'][0][0] if len(x['opportunity']) else None
+                ))
+            | 'Group by key' >> beam.GroupByKey()
+            | 'Combine the stats' >> beam.CombineValues(CombineStats())
+            | 'Generate models from stats' >> beam.MapTuple(
+                self._generate_translation_contribution_model)
+        )
```

Comment:
This code is not updated, please check the current cron_jobs.py

Reply:
Done.